automation_tools/satellite6/upgrade/client.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+from automation_tools.repository import disable_repos
+from automation_tools.satellite6.upgrade.tasks import docker_execute_command
+from fabric.api import execute, run
+from tasks import (
+    generate_satellite_docker_clients_on_rhevm,
+    refresh_subscriptions_on_docker_clients,
+    remove_all_docker_containers,
+    sync_tools_repos_to_upgrade
+)
+
+
+def personal_clients_upgrade(old_repo, clients):
+    """Helper function to run upgrade on personal clients
+
+    :param string old_repo: The old tools repo to disable before updating
+        katello-agent package
+    :param list clients: The list of clients onto which katello-agent package
+        will be updated
+    """
+    for client in clients:
+        print('Upgrading client on docker container: {}'.format(client))
+        print('Previous katello-agent version:')
+        execute(lambda: run('rpm -qa | grep katello-agent'), host=client)
+        print('Disabling existing client tools repo:')
+        execute(disable_repos, old_repo, host=client)
+        print('Upgrading Clients:')
+        execute(lambda: run('yum update katello-agent'), host=client)
+        print('katello-agent package after installation:')
+        execute(lambda: run('rpm -qa | grep katello-agent'), host=client)
+
+
+def docker_clients_upgrade(old_repo, container_ids):
+    """Helper function to run upgrade on docker containers as clients
+
+    :param string old_repo: The old tools repo to disable before updating
+        katello-agent package
+    :param list container_ids: The list of container_ids onto which
+        katello-agent package will be updated
+    """
+    for client in container_ids:
+        print('Upgrading client on docker container: {}'.format(client))
+        print('Previous katello-agent version:')
+        docker_execute_command(client, 'rpm -qa | grep katello-agent')
+        print('Disabling existing client tools repo:')
+        docker_execute_command(
+            client, 'subscription-manager repos --disable {}'.format(old_repo))
+        print('Upgrading clients on Docker Containers')
+        docker_execute_command(client, 'yum update katello-agent')
+        print('katello-agent package after installation:')
+        docker_execute_command(client, 'rpm -qa | grep katello-agent')
+
+
+def satellite6_client_setup():
+    """Sets up required things on upgrade running machine and on Client to
+    perform client upgrade later
+
+    If not personal clients, then it creates docker containers as clients on
+    rhevm vm.
+    """
+    # If Personal Client Hostname provided
+    if os.environ.get('CLIENT6_HOSTS') or os.environ.get('CLIENT7_HOSTS'):
+        clients6 = list(os.environ.get('CLIENT6_HOSTS'))
+        clients7 = list(os.environ.get('CLIENT7_HOSTS'))
+        # Sync latest sat tools repo to clients if downstream
+        if all([
```

Comment:
Is these env vars required when downstream? If yes the code should check for them if not doing already.